src/main/scala/mrtjp/projectred/expansion/TileProjectBench.scala

```diff
@@ -186,123 +221,382 @@ class SlotProjectCrafting(
     idx: Int,
     x: Int,
     y: Int
-) extends SlotCrafting(player, tile.invCrafting, tile.invResult, idx, x, y)
+) extends SlotCrafting(
+      player,
+      tile.invCrafting,
+      tile.invCraftingResult,
+      idx,
+      x,
+      y
+    )
     with TSlot3 {
+
+  private val previous_indices = Array[Int](
```

Comment:
This part can be weird, but it will be removed in future, still bad at `scala` syntax